a.el

```diff
@@ -124,10 +126,12 @@
     (length coll))
 
    ((hash-table-p coll)
-    (length (a-keys coll)))))
+    (hash-table-count coll))))
 
+;; 1) is it ok that collections of different types can be equal?
+;; 2) early termination would be nice for larger collections
```

Comment:
That's true. The `(and bool ,,,)` short circuits so you skip any remaining value lookups and equality checks, but it'll still go through the rest of the loop.

In general I haven't been too concerned with performance (yet), assuming that if you're doing this style of programming performance isn't your main concern either, but performance improvements are of course very welcome.